File: android/app/src/main/java/com/nishant/mast/Allergen.java
package com.nishant.mast;

import java.io.Serializable;

/**
 * Created by Nishant on 7/24/17.
 */

public class Allergen implements Serializable {

    public String name;
    public String source;
    public String order;
    public String species;
    public String bName;
    public String mw;
    public String aIcity;
    public String aRef;
    public String food;
    public String pdbID;
    public String category;
    public boolean sold;
 
    public Allergen(String name, String source, String order,
                    String species, String bName, String mw,
                    String aIcity, String aRef, String food,
                    String pdbID, String category, boolean sold) {
        this.name = name;
        this.source = source;
        this.order = order;
        this.species = species;
        this.bName = bName;
        this.mw = mw;
        this.aIcity = aIcity;
        this.aRef = aRef;
        this.food = food;
        this.pdbID = pdbID;
        this.category = category;
        this.sold = sold;
    }

    @Override
    public String toString() {
        return "Allergen{" +
                "name='" + name + '\'' +
                ", source='" + source + '\'' +
                ", order='" + order + '\'' +
                ", species='" + species + '\'' +
                ", bName='" + bName + '\'' +
                ", mw='" + mw + '\'' +
                ", aIcity='" + aIcity + '\'' +
                ", aRef='" + aRef + '\'' +
                ", food='" + food + '\'' +
                ", pdbID='" + pdbID + '\'' +
                ", sold='" + sold + '\'' +
                ", category='" + category + '\'' +
                '}';
    }
}

File: ios/Mast/Allergen.swift
//
//  Allergen.swift
//  Mast
//
//  Created by Nishant Jha on 10/15/17.
//  Copyright © 2017 Nishant Jha. All rights reserved.

import Foundation

class Allergen {
    
    var name: String = ""
    var source: String = ""
    var order: String = ""
    var species: String = ""
    var bName: String = ""
    var mw: String = ""
    var aIcity: String = ""
    var aRef: String = ""
    var food: String = ""
    var PDBid: String = ""
    var sold : Bool = false
    var category: String = "None"
    
    public var description: String {
        return self.name + ", " + self.PDBid
    }
    
    init(name: String, source: String, order: String, species: String,
         bName: String, mw: String, aIcity: String, aRef: String,
         food: String, PDBid: String, sold: Bool, category: String) {
        
        self.name = name;
        self.source = source;
        self.order = order;
        self.species = species;
        self.bName = bName;
        self.mw = mw;
        self.aIcity = aIcity;
        self.aRef = aRef;
        self.food = food;
        self.PDBid = PDBid;
        self.sold = sold;
        if(category.contains("None")) {
            self.category = "Other,"
        }
        else {
            self.category = category
        }
    }
    
    init() {
        self.name = "Nishant";
        self.source = "Jha";
        self.order = "Mage";
        self.species = "Human";
        self.bName = "Homo Sapiens";
        self.mw = "300 kDa";
        self.aIcity = "Yup!";
        self.aRef = "www.zombo.com";
        self.food = "Yes";
        self.PDBid = "HELP"
        self.sold = true;
        self.category = "None"
    }
}

